tendermint/src/abci/transaction/hash.rs

```diff
@@ -28,6 +29,15 @@ impl Hash {
     }
 }
 
+impl Default for Hash {
+    fn default() -> Hash {
+        // hash of empty data
+        let mut bytes = [0u8; LENGTH];
```

Comment:
A `Hash` of all zero bytes is generally not semantically meaningful and IMO should not be the `Default`.

(I have seen security vulnerabilities arising from zero-value hashes in Go)

Reply:
Makes sense, just removed.